testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -689,6 +697,34 @@ def test_spherical_layer(self, u, periodic):
 
         assert ref == set(result.indices)
 
+    @pytest.mark.parametrize('periodic', (True, False))
+    def test_isolayer(self, u, periodic):
+
+        rmin, rmax = 2.4, 3
+        if u.atoms.types[0].isdigit():
+            r1 = u.select_atoms("resid 1 or resid 2 and type 7")
+        else:
+            r1 = u.select_atoms("resid 1 or resid 2 and type O")
+
+        sel = Parser.parse("isolayer {} {} (index {} or index {})".format(rmin, rmax, *r1.ix), u.atoms)
+        if periodic:
+            sel.periodic = True
+        else:
+            sel.periodic = False
+        result = sel.apply(u.atoms)
+
+        box = u.dimensions if periodic else None
+        cog1 = r1[0].position.reshape(1, 3)
+        d1 = distance_array(u.atoms.positions, cog1, box=box)
+        cog2 = r1[1].position.reshape(1, 3)
+        d2 = distance_array(u.atoms.positions, cog2, box=box)
+
+        ref_inner  = set(np.where((d1 < rmin) | (d2 < rmin))[0])
+        ref_outer  = set(np.where((d1 < rmax) | (d2 < rmax))[0])
+        ref_outer -= ref_inner
+
+        assert ref_outer == set(result.indices)
```

Comment:
just to check, this selection is capturing something right?

Reply:
Hey! Yes, there are about 12 or so atoms. Do you want me to add something with the length?